qiskit_optimization/translators/docplex.py

```diff
@@ -0,0 +1,288 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2021.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""Model translators between QuadraticProgram and Docplex"""
+
+from typing import cast, Any
+
+from docplex.mp.constr import LinearConstraint as DocplexLinearConstraint
+from docplex.mp.constr import NotEqualConstraint
+from docplex.mp.constr import QuadraticConstraint as DocplexQuadraticConstraint
+from docplex.mp.model import Model
+from docplex.mp.quad import QuadExpr
+from docplex.mp.vartype import BinaryVarType, ContinuousVarType, IntegerVarType
+
+try:
+    # new location since docplex 2.16.196
+    from docplex.mp.dvar import Var
+except ImportError:
+    # old location until docplex 2.15.194
+    from docplex.mp.linear import Var
+
+from qiskit_optimization.exceptions import QiskitOptimizationError
+from qiskit_optimization.problems.constraint import Constraint
+from qiskit_optimization.problems.quadratic_objective import QuadraticObjective
+from qiskit_optimization.problems.variable import Variable
+from .model_translator import ModelTranslator
+
+
+class DocplexTranslator(ModelTranslator):
+    """Translator between a Docplex model and a quadratic program
+    """
+
+    def qp_to_model(self, quadratic_program: Any) -> Any:
```

Comment:
Should we have correct type hints?

Reply:
It is impossible due to pylint outputs cyclic import errors.